src/SIL.XForge.Scripture/ClientApp/src/app/translate/training-progress/training-progress.component.ts

```diff
@@ -87,6 +91,14 @@ export class TrainingProgressComponent extends DataLoadingComponent implements O
     this.trainingProgressClosed = true;
   }
 
+  private get canEditTexts(): boolean {
+    const project = this.projectDoc?.data;
+    return (
+      project != null &&
+      SF_PROJECT_RIGHTS.hasRight(project, this.userService.currentUserId, SFProjectDomain.Texts, Operation.Edit)
```

Comment:
What if the user has edit permission, but not for this text? Permission to edit can be at the chapter level, and the rights service will return true if the user has a general right to edit. See the editor component.

Reply:
Good question. As this component only shows training status, not suggestions, it is appropriate to check for the general edit permission (the same as is checked on the Translation Overview screen). If someone was editing a chapter they had permission to view, navigated to the Translation Overview screen, then navigated to a chapter they could not edit, the Training Status should still display while training is in progress.

For clarity, I have renamed this method to `userHasGeneralEditRight`, matching `editor.component.ts`.

Reply:
Ah, right, somehow I misunderstood the context we were dealing with.